File: examples/adhoc/AdhocPush.java
import java.util.Random;

public class AdhocPush {
    public static boolean isPrime(long n) {
        for (long i = 2; i <= n; i++) {
            if (i * i > n) return true;
            if (n % i == 0) return false;
        }
        return false;
    }

    public static long slow(long n) {
        long sum = 0;
        for (long i = 0; i <= n; i++) {
            sum += i;
        }
        return sum;
    }

    public static long fast(long n) {
        long sum = 0;
        long root = (long) Math.sqrt(n);
        for (long a = 1; a <= n; a += root) {
            long b = Math.min(a + root -1, n);
            sum += (b - a + 1) * (a + b) / 2;
        }
        return sum;
    }

    public static void run() {
        Random r = new Random();
        long base = r.nextInt(1000000) + 1;
        for (long i = 0; i < 40000000; i++) {
            long n = r.nextInt(10000) + 1;
            if (isPrime(base + i))
                slow(n);
            else
                fast(n);
        }
    }

    public static void main(String[] args) {
        run();
    }
}

File: examples/adhoc/adhoc-push.py
#!/usr/bin/env python
from random import randint
from math import sqrt

import pyroscope


def is_prime(n):
    for i in range(2, n):
        if i * i > n:
            return True
        if n % i == 0:
            return False
    return False


def slow(n):
    sum = 0
    for i in range(1, n+1):
        sum += i
    return sum


def fast(n):
    sum = 0
    root = int(sqrt(n))
    for a in range(1, n+1, root):
        b = min(a + root - 1, n)
        sum += (b - a + 1) * (a + b) // 2
    return sum


def run():
    base = randint(1, 10**6)
    for i in range(2*10**6):
        [fast, slow][is_prime(base + i)](randint(1, 10**4))


if __name__ == '__main__':
    # No need to modify existing settings,
    # pyroscope will override the server address
    pyroscope.configure(
        app_name="adhoc.push.python",
        server_address="http://pyroscope:4040",
    )
    run()
